src/ape_node/provider.py

```diff
@@ -405,7 +409,10 @@ def restore(self, snapshot_id: SnapshotID):
             logger.error("Unable to set head to future block.")
             return
 
-        self.make_request("debug_setHead", [block_number_hex_str])
+        try:
+            self.make_request("debug_setHead", [block_number_hex_str])
+        except Exception:
+            self._supports_debug_set_head = False
```

Comment:
Should this raise `NotImplementedError` instead at this point?

Reply:
that is actually the exception that was already be raised!

We could move the handling `chain.snapshot` though: https://github.com/ApeWorX/ape/blob/main/src/ape/managers/chain.py#L1607